src/compiler/debug.rs

```diff
@@ -1,12 +1,43 @@
 #![allow(non_upper_case_globals)]
 
+use std::borrow::Borrow;
+use std::cell::Cell;
 use std::fmt;
 
-use crate::NodeArray;
+use crate::{AssertionLevel, Node, NodeArray};
+
+enum AssertionKeys {
+    AssertNode,
+}
+
+thread_local! {
+    static current_assertion_level: Cell<AssertionLevel> = Cell::new(AssertionLevel::None);
```

Comment:
Making this a struct member of `DebugType` would've made it non-thread-safe and I didn't want to make the ergonomics of using `Debug_` worse (eg by converting it to a `thread_local!`) so just made it a "standalone" global static